tianshou/data/batch.py

```diff
@@ -534,21 +538,23 @@ def __len__(self) -> int:
         return min(r)
 
     @property
-    def size(self) -> int:
-        """Return self.size."""
+    def shape(self) -> List[int]:
+        """Return self.shape."""
         if len(self.__dict__.keys()) == 0:
-            return 0
+            return []
         else:
-            r = []
+            data_shape = []
             for v in self.__dict__.values():
-                if isinstance(v, Batch):
-                    r.append(v.size)
-                elif hasattr(v, '__len__') and (not isinstance(
-                        v, (np.ndarray, torch.Tensor)) or v.ndim > 0):
-                    r.append(len(v))
-                else:
-                    r.append(1)
-            return min(r) if len(r) > 0 else 0
+                try:
+                    if len(v.shape) > 0:
+                        data_shape.append(v.shape)
+                except AttributeError:
+                    if isinstance(v, list):
+                        data_shape.append(tuple([len(v)]))
+                    else:
+                        raise TypeError("No support for 'shape' method with "
```

Comment:
Add would rather use `__setitem__` and `__setattr__` to make sure that `list` are converted to numpy, so that this check can be avoided.